src/api/rest/providers/github/schemas/users.js

```diff
@@ -0,0 +1,89 @@
+import { schema } from 'normalizr';
+import moment from 'moment/min/moment.min';
+
+export const userSchema = new schema.Entity(
+  'users',
+  {},
+  {
+    idAttribute: user => user.login.toLowerCase(),
+    processStrategy: entity => {
+      const processed = {};
+
+      // These are provided in both mini & full modes
+      processed.id = entity.login.toLowerCase(); // id should be always used for navigation
+      processed.login = entity.login;
+      processed.avatarUrl = entity.avatar_url;
+
+      // These flags should be in all our schemas.
+      processed._isComplete = false; // entity not fully fetched yet
+      processed._isAuth = false; // entity doesn't belong to the auth user
+      processed._entityUrl = false; // The github url for the entity. To be used in openInBrowser()
+      processed._fetchedAt = moment().format('X');
+
+      // name is only present in full mode, we base our full parsing on its presence
+      if (typeof entity.name !== 'undefined') {
```

Comment:
Same here. Pretty much anywhere with `undefined` checks, can we not just simplify them to like in this case `if(entity.name){...}`

Reply:
This can be tricky and error prone in a schema.

If the typeof *is* undefined, then the field wasn't available in the original JSON. We use that to determine what form of the entity we received from the API (complete, minimal, etc)

If for some reason entity.name is provided, but as an empty string, the test will not do what we want.

I'll add a `isProvided()` helper to make things clearer and have only one typeof undefined check.